### DISCUSSION TITLE
Bad character encoding in snapshots

### DISCUSSION THREAD

**Discussion by prsdta** [percy/cli#721] [Support] [answered]
Hello, we've been having a problem with a Percy snapshot recently, and I'm having a hard time finding a solution.

Rather randomly but regularly, one of our snapshots fails due to characters being garbled.
|normal | error|
|-------|------|
| <img width="172" alt="image" src="https://user-images.githubusercontent.com/19373361/149434888-fb7b864a-b0e5-4877-8553-b3f42ed59c73.png"> | <img width="184" alt="image" src="https://user-images.githubusercontent.com/19373361/149434917-71eed4bd-5f6e-40f4-93f9-3408dd4f318b.png"> <img width="172" alt="image" src="https://user-images.githubusercontent.com/19373361/149435055-4333477f-e072-451e-bab4-4de3d79435ce.png"> |

Only this text node fails, and always in the same two patterns. All our text is UTF-8, and no other snapshot fails for this reason. The node uses the same font as adjacent ones.

We're using percy-capybara + percy/cli.

If you have any suggestions on how to solve this issue I'd be grateful.

**Comment by Robdel12** [comment, marked as answer]
Hey @prsdta! Every time I've seen this it's due to a missing charset in the `<head>`. This should help: https://docs.percy.io/docs/debugging-sdks#bad-character-encoding

Since it's flakey, it's likely there's something in the app that's putting invalid HTML into the `<head>`. When that rerenders in Percy, the browser will "fix" it, by forcing it all out into the `<body>`

**Reply by prsdta** [reply]
Thanks for the answer!
I couldn't find anything problematic in my code but I'm starting to suspect 3rd party scripts. I'll keep looking.

**Reply by Robdel12** [reply]
If you pass over a snapshot ID I can look at the snapshot and see what's going on 👀. 

The snapshot ID is`2222222` in this example URL: `percy.io/example-org/example-project/builds/1111111/changed/2222222`
